File: src/main/java/kingdoms/server/ServerProxy.java
package kingdoms.server;

import kingdoms.server.commands.CommandTOK;
import kingdoms.server.handlers.*;
import kingdoms.server.items.ItemCoin;
import kingdoms.server.items.ItemSpawnEgg;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.event.FMLServerStartingEvent;
import cpw.mods.fml.common.registry.GameRegistry;
import net.minecraft.command.CommandHandler;
import net.minecraft.item.Item;

public class ServerProxy
{
    private Config config;
    public static final Item coins = new ItemCoin().setUnlocalizedName("Coins");

    public void pre(FMLPreInitializationEvent e)
    {
        config = new Config(e);
        EntitiesRegister.INSTANCE.register();

        GameRegistry.registerItem(coins, "Coins");
        GameRegistry.registerItem(new ItemSpawnEgg(), "MonsterPlacer");

        new ServerEvents();
        new NetworkHandler();
    }

    public void init(FMLInitializationEvent e)
    {
        new GuiHandler();
    }

    public void starting(FMLServerStartingEvent e)
    {
        ((CommandHandler) e.getServer().getCommandManager()).registerCommand(new CommandTOK());
    }

    public Config getConfig()
    {
        return config;
    }
}

File: src/main/java/kingdoms/server/ServerProxy.kt
package kingdoms.server

import cpw.mods.fml.common.event.FMLInitializationEvent
import cpw.mods.fml.common.event.FMLPreInitializationEvent
import cpw.mods.fml.common.event.FMLServerStartingEvent
import cpw.mods.fml.common.registry.GameRegistry
import kingdoms.server.commands.CheatCodes
import kingdoms.server.handlers.*
import kingdoms.server.items.ItemCoin
import kingdoms.server.items.ItemSpawnEgg
import kingdoms.server.world.WorldGen
import net.minecraft.command.CommandHandler
import net.minecraft.item.Item

open class ServerProxy
{
    lateinit var config: Config

    companion object
    {
        val coins: Item = ItemCoin().setUnlocalizedName("Coins")
    }

    open fun pre(e: FMLPreInitializationEvent)
    {
        config = Config(e)
        EntitiesRegister.register()

        GameRegistry.registerItem(coins, "Coins")
        GameRegistry.registerItem(ItemSpawnEgg(), "MonsterPlacer")

        ServerEvents()
        NetworkHandler

        GameRegistry.registerWorldGenerator(WorldGen(), 1000)
    }

    open fun init(e: FMLInitializationEvent)
    {
        GuiHandler()
    }

    open fun starting(e: FMLServerStartingEvent)
    {
        val command = e.server.commandManager as CommandHandler

        command.registerCommand(CheatCodes())
    }
}